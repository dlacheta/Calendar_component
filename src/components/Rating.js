import React from "react";
import PropTypes from "prop-types";

const Rating = ({ rating, ratingAmount }) => {
  return (
    <div className="rating">
      <i className={rating >= 1 ? "material-icons full" : "material-icons"}>
        star_rate
      </i>
      <i className={rating >= 2 ? "material-icons full" : "material-icons"}>
        star_rate
      </i>
      <i className={rating >= 3 ? "material-icons full" : "material-icons"}>
        star_rate
      </i>
      <i className={rating >= 4 ? "material-icons full" : "material-icons"}>
        star_rate
      </i>
      <i className={rating >= 5 ? "material-icons full" : "material-icons"}>
        star_rate
      </i>
      {ratingAmount}
    </div>
  );
};

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
  ratingAmount: PropTypes.number.isRequired
};

export default Rating;
