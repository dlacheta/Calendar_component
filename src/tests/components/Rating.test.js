import React from "react";
import { shallow } from "enzyme";
import Rating from "../../components/Rating";

it("Rating component renders correctly", () => {
  const rating = 4;
  const ratingAmount = 123;
  const wrapper = shallow(
    <Rating rating={rating} ratingAmount={ratingAmount} />
  );

  expect(wrapper).toMatchSnapshot();
});
