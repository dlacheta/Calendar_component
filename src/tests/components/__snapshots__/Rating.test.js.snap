// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rating
    rating={4}
    ratingAmount={123}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          className="material-icons full"
        >
          star_rate
        </i>,
        <i
          className="material-icons full"
        >
          star_rate
        </i>,
        <i
          className="material-icons full"
        >
          star_rate
        </i>,
        <i
          className="material-icons full"
        >
          star_rate
        </i>,
        <i
          className="material-icons"
        >
          star_rate
        </i>,
        123,
      ],
      "className": "rating",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "star_rate",
          "className": "material-icons full",
        },
        "ref": null,
        "rendered": "star_rate",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "star_rate",
          "className": "material-icons full",
        },
        "ref": null,
        "rendered": "star_rate",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "star_rate",
          "className": "material-icons full",
        },
        "ref": null,
        "rendered": "star_rate",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "star_rate",
          "className": "material-icons full",
        },
        "ref": null,
        "rendered": "star_rate",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "star_rate",
          "className": "material-icons",
        },
        "ref": null,
        "rendered": "star_rate",
        "type": "i",
      },
      123,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="material-icons full"
          >
            star_rate
          </i>,
          <i
            className="material-icons full"
          >
            star_rate
          </i>,
          <i
            className="material-icons full"
          >
            star_rate
          </i>,
          <i
            className="material-icons full"
          >
            star_rate
          </i>,
          <i
            className="material-icons"
          >
            star_rate
          </i>,
          123,
        ],
        "className": "rating",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "star_rate",
            "className": "material-icons full",
          },
          "ref": null,
          "rendered": "star_rate",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "star_rate",
            "className": "material-icons full",
          },
          "ref": null,
          "rendered": "star_rate",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "star_rate",
            "className": "material-icons full",
          },
          "ref": null,
          "rendered": "star_rate",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "star_rate",
            "className": "material-icons full",
          },
          "ref": null,
          "rendered": "star_rate",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "star_rate",
            "className": "material-icons",
          },
          "ref": null,
          "rendered": "star_rate",
          "type": "i",
        },
        123,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
